web/server/api/endpoints/files.py

```diff
@@ -55,22 +63,38 @@ async def write_file(
     path: str = Depends(validate_path),
     settings: Settings = Depends(get_settings),
     context: Context = Depends(get_context),
-    path_mapping: t.Dict[Path, models.FileType] = Depends(get_path_mapping),
 ) -> models.File:
     """Create, update, or rename a file."""
     path_or_new_path = path
     if new_path:
         path_or_new_path = validate_path(new_path, context)
         replace_file(settings.project_path / path, settings.project_path / path_or_new_path)
     else:
-        (settings.project_path / path_or_new_path).write_text(content, encoding="utf-8")
-
+        full_path = settings.project_path / path
+        is_sql = pathlib.Path(path_or_new_path).suffix == ".sql"
+        path_to_model_mapping = await get_path_to_model_mapping(settings=settings)
+        model = path_to_model_mapping.get(Path(path_or_new_path))
+        default_dialect = context.config_for_path(Path(path_or_new_path)).dialect
+        dialect = model.dialect if model and model.is_sql else default_dialect
+        if is_sql:
+            try:
+                expressions = parse(content, default_dialect=default_dialect)
+                content = format_model_expressions(expressions, dialect)
+            except Exception:
+                ApiException(
+                    message="Unable to format SQL file",
+                    origin="API -> files -> write_file",
+                )
```

Comment:
This can be removed or replaced with a SSE message warning the file couldn't be formatted. What do you think?